include/circt/Dialect/HW/HWOpInterfaces.td

```diff
@@ -107,41 +107,46 @@ def HWMutableModuleLike : OpInterface<"HWMutableModuleLike"> {
       return getModulePortInfo($_op);
     }]>,
 
-    /// Insert and remove input and output ports of this module. Does not modify
-    /// the block arguments of the module body. The insertion and removal
-    /// indices must be in ascending order. The indices refer to the port
-    /// positions before any insertion or removal occurs. Ports inserted at the
-    /// same index will appear in the module in the same order as they were
+    /// Insert and remove input and output ports of this module. Modifies the
+    /// body block arguments if 'body' is non-null to match the new inputs. Does
+    /// _not_ modify the terminator to match the outputs. The insertion and
+    /// removal indices must be in ascending order. The indices refer to the
+    /// port positions before any insertion or removal occurs. Ports inserted at
+    /// the same index will appear in the module in the same order as they were
     /// listed in the insertion arrays.
     InterfaceMethod<"Insert and remove input and output ports",
     "void", "modifyPorts", (ins
```

Comment:
The reason to have it in the OpInterface is so that it works on all ops which implement the op interface. hw::modifyModulePorts only works on HWModuleOps and other ops which just happen to be compatible. That's the point of an op interface! Same reason you don't want to use the helper method.